plotting/basemap.py

```diff
@@ -1,120 +1,98 @@
 import hashlib
-import math
 import os
 import pickle
 import threading
 
-from cachetools import LRUCache
+import cartopy.crs as ccrs
+import cartopy.io.shapereader as shpreader
+import matplotlib.pyplot as plt
 from flask import current_app
-from mpl_toolkits.basemap import Basemap
+from typing import Union
 
-from utils import return_type
 
-_maps_cache = LRUCache(maxsize=64)
+def get_resolution(height: float, width: float) -> str:
+    area = height * width / 1e6
+
+    if area < 1e8:
+        return "f"  # full resolution
+    elif area < 1e12:
+        return "i"  # intermediate resolution
+    else:
+        return "c"  # crude resolution
 
 
-def _convert_to_bounded_lon(lon: float) -> float:
+def _get_shapefile(resolution: str) -> shpreader.BasicReader:
+
+    if resolution == "f":
+        land_shp = shpreader.BasicReader(
+            current_app.config["SHAPE_FILE_DIR"] + "ne_10m_diff.shp"
+        )
+    elif resolution == "i":
+        land_shp = shpreader.BasicReader(
+            current_app.config["SHAPE_FILE_DIR"] + "ne_50m_diff.shp"
+        )
+    else:
+        land_shp = shpreader.BasicReader(
+            current_app.config["SHAPE_FILE_DIR"] + "ne_110m_diff.shp"
+        )
 
-    return math.fmod(lon, 360.0)
+    return land_shp
 
 
 def load_map(
-    projection: str, center: tuple, height: float, width: float, min_lat: float = 0
-) -> Basemap:
+    plot_proj: ccrs, extent: list, figuresize: list, dpi: int, plot_resolution: str
+) -> Union[plt.figure, plt.axes]:
 
     CACHE_DIR = current_app.config["CACHE_DIR"]
-    filename = _get_filename(projection, center, height, width)
+    filename = _get_filename(plot_proj.proj4_params["proj"], extent)
+    filename = "".join([CACHE_DIR, "/", filename])
 
-    def get_resolution(h: float, w: float) -> str:
-        area_km = (h * w) / (1000 * 1000)
+    land_shp = _get_shapefile(plot_resolution)
 
-        if area_km < 10000:
-            return "f"  # full resolution
-        elif area_km < 100000:
-            return "h"  # high resolution
-        elif area_km < 1000000:
-            return "i"  # intermediate resolution
-        elif area_km < 10000000:
-            return "l"  # low resolution
-        else:
-            return "c"  # crude resolution
-
-    if _maps_cache.get(filename) is None or True:
-        filename = "".join([CACHE_DIR, "/", filename])
```

Comment:
It looks like we were changing the filename here before adding it to the _maps_cache but that prevented us from ever retrieving it. Moving the filename change to before this check allows us to grab the plot from the cache but it always returned an empty axis so I ended up scrapping it and using the pickled plot instead.